### DISCUSSION TITLE
为什么总是需要无意义的 ID · Why's THE Design? · /whys-the-design-meaningless-identifier

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#578] [Announcements]
https://draveness.me/whys-the-design-meaningless-identifier

**Comment by flyisland** [comment]
难点是又要无意义，又要全局唯一

**Comment by draveness** [comment]
> 难点是又要无意义，又要全局唯一

只要『没意义』就可以全局唯一

**Comment by flyisland** [comment]
> > 难点是又要无意义，又要全局唯一
> 
> 只要『没意义』就可以全局唯一

我的意思是有多个ID生成器为前提。为了达到全局唯一，不同的ID生成器之间总会分区吧，这个”区“就具备一定意义了。或者有什么好的算法请不吝指教。

**Reply by PeterlitsZo** [reply]
雪花算法？

**Comment by tldzyx** [comment]
> @draveness
> > 难点是又要无意义，又要全局唯一
> 
> 只要『没意义』就可以全局唯一

没意义不代表不会重复, 实际场景需要的是 全局唯一标识符, 只要全局唯一了, 有意义或者无意义其实无所谓. 

文章想表达的是: 在 有限 的空间内, 有意义的数据 占据 了表示 唯一标识符 的可用空间, 导致唯一标识符 溢出.

-|全局唯一|不全唯一
---|---|---
**有意义**|理想|冲突
**无意义**|妥协|无用
